test/integration2/util/knex-instance-provider.js

```diff
@@ -208,8 +249,44 @@ function getKnexForDb(db, configOverrides = {}) {
   });
 }
 
+/** @returns {import('../../../types/index').Knex} */
+function getKnexForSqlite(foreignKeys, configOverrides = {}) {
+  const sql = `PRAGMA foreign_keys = ${foreignKeys ? 'ON' : 'OFF'}`;
+  const config = {
+    ...testConfigs.sqlite3,
+    ...configOverrides,
+  };
+  config.pool = {
+    ...config.pool,
+    afterCreate: function (connection, callback) {
+      connection.run(sql, callback);
+    },
+  };
+  return knex(config);
+}
+
+/** @returns {import('../../../types/index').Knex} */
+function getKnexForBetterSqlite(foreignKeys, configOverrides = {}) {
+  const sql = `PRAGMA foreign_keys = ${foreignKeys ? 'ON' : 'OFF'}`;
+  const config = {
+    ...testConfigs.sqlite3,
+    ...configOverrides,
+  };
+  config.pool = {
+    ...config.pool,
+    afterCreate: function (connection, callback) {
+      connection.prepare(sql).run();
+      callback(null, connection);
+    },
+  };
+  return knex(config);
+}
+
```

Comment:
These functions parameterize the sqlite knex instance so that connections are initialized with either foreign key checking on or off. This is in support of the new tests that are added; this could be integrated into the other bits, but I wanted to keep this PR as self-contained as possible and not also involve refactoring everything.